### PR TITLE
i18n: Load user translations

### PR DESCRIPTION
**Pull request by deBhal**
#### Changes proposed in this Pull Request

This PR adds the option to load and display any user's waiting translations in place of the current accepted ones.

#### Testing instructions

Add a `?load-user-translations=username` query parameter when loading calypso to load that user's waiting translations.  You can specify optional `locale`, `project` and `translationSet` arguments (so you can load waiting strings from the test project, de_formal, or even a different locale).

Translations___French___WordPress_com___GlotPress_-_julesaus_dev_dfw_wordpress_com_and_Planes_‹_julesauspremiumtest_—_WordPress_com

Logged out: (e.g. http://calypso.localhost:3000/start/user/fr?load-user-translations=julesaus&project=test )

Translations___French__France____Test___GlotPress_-_julesaus_dev_dfw_wordpress_com_and_Créer_un_site_—_WordPress_com_and___a8c_wp-calypso_client_blocks_signup-form_index_jsx_—_wp-calypso

Happily, we already transfer the params across, so user translations are reloaded if you click on e.g. "Également disponible en Français" (also available in)

### CODE REVIEW

**Comment by deBhal** [comment]
I had thought that the query parsing belongs in the routing logic, but I believe I was wrong.

The existing page middleware does parse the search string as well as determine the appropriate locale information, but it stores it immediately in the redux store a nd persists it indefinitely, which is appropriate - a language choice is relevant until a new one is made.

The intention here is much more temporary - we expect these strings to 
be loaded up for specific short-term purposes, like seeing how new strings look be fore being added to the core set and deployed normally.

It is potentially a large amount of additional data, so if we decide to start usin g this to use this for longer term purposes we should add some caching for the dat a and reconsider adding the parameters it to the redux tree alongside the current locale data.

File: client/lib/i18n-utils/switch-locale.js [hunk 1]
```diff
@@ -107,10 +108,62 @@ export default function switchLocale( localeSlug ) {
...
+		'projects',
+		project,
+		locale,
+		translationSet,
+		'export-translations',
+	].join( '/' );
+
+	const query = {
+		'filters[user_login]': username,
+		'filters[status]': 'waiting',
```
Comment (akirk) [inline client/lib/i18n-utils/switch-locale.js]: I think we'll need to use `current_or_waiting_or_fuzzy_or_untranslated` for now. The main use case is actually for reviewers to be able to disconnect from i18n-deploy, so their translations will be `current` thus not included in this query.

**Comment by akirk** [comment]
I've taken the liberty and change the default `filter[status]` and allow changing the parameter. Ideally, loading `current` translations would not need a username check, any other status should only allow to load your own translations in order to prevent purposeful wrong translation insertion.

**Comment by deBhal** [comment]
Yeah, loading current strings from any user should be safe, but we do need a bit more care with waiting translations.

**Comment by akirk** [comment]
Unfortunately this doesn't seem to work on production, could you check?

**Pull request merged**